packages/components/app/styles/components/table.scss

```diff
@@ -119,23 +119,31 @@ $hds-table-cell-padding-tall: 20px 16px;
 }
 
 // TABLE DENSITY (TBODY ROW HEIGHT)
+.hds-table--density-short .hds-table__tbody th,
 .hds-table--density-short .hds-table__tbody td {
+  height: unset;
   padding: $hds-table-cell-padding-short;
 }
 
+.hds-table--density-medium .hds-table__tbody th,
 .hds-table--density-medium .hds-table__tbody td {
+  height: unset;
```

Comment:
instead of unsetting the height here (which may lead to css specificity battles with the consumers, better to change where the height is set (above) and contextualize the declaration to the `.hds-table__thead` selector